### PR TITLE
feat(config): Add additional validation checks and defaults

### LINKED ISSUE #2: Add TOML config package
In order to make mocktimism easy to configure we want to offer a toml file configuration option

- [x] Add toml parsing with https://github.com/BurntSushi/toml
- [x] Add validation
- [x] Add tests
- [x] Document the toml 

image

### PR DESCRIPTION
**Pull request by roninjin10**
Add additional checks and defaults for config validation and loading

fixes https://github.com/ethereum-optimism/mocktimism/issues/2
Work towards https://github.com/ethereum-optimism/mocktimism/issues/24

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* main
  * **PR #69** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/69" target="_blank">Graphite</a> 
    * **PR #70** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/70" target="_blank">Graphite</a> 
      * **PR #71** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/71" target="_blank">Graphite</a> 
        * **PR #74** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/74" target="_blank">Graphite</a>  👈
          * **PR #75** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/75" target="_blank">Graphite</a> 
            * **PR #76** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/76" target="_blank">Graphite</a> 
              * **PR #78** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/78" target="_blank">Graphite</a> 
                * **PR #80** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/80" target="_blank">Graphite</a> 
                  * **PR #81** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/81" target="_blank">Graphite</a> 
                    * **PR #82** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/82" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/mocktimism/74?utm_source=stack-comment).

File: cmd/action_config.go [hunk 1]
```diff
@@ -14,9 +15,10 @@ import (
 func actionConfig(ctx *cli.Context) error {
 	log := oplog.NewLogger(oplog.AppOut(ctx), oplog.ReadCLIConfig(ctx)).New("role", "mocktimism")
 	oplog.SetGlobalLogHandler(log.GetHandler())
-	cfg, err := config.LoadNewConfig(log, ctx.String(ConfigFlag.Name))
-	if err != nil {
-		log.Error("failed to load config", "err", err)
+	cfg, errs := config.LoadNewConfig(log, ctx.String(ConfigFlag.Name))
+	if len(errs) > 0 {
+		err := errors.Join(errs...)
```
Comment (hamdiallam) [inline cmd/action_config.go]: you should push `errors.Join` into `config.LoadNewConfig` to keep the `if err != nil` pattern. return a list of errors feels unorthodox

**Review by hamdiallam — APPROVED** [review]

File: cmd/action_config.go (hunk 1 above)
Comment (roninjin10) [reply cmd/action_config.go]: Ah yea didn't think about this but I agree

Comment (roninjin10) [reply cmd/action_config.go]: Done

**Pull request merged**